tendermint-rs/src/amino_types/proposal.rs

```diff
@@ -129,6 +130,25 @@ impl SignableMsg for SignProposalRequest {
             None => Err(MissingConsensusMessage.into()),
         }
     }
+    fn consensus_state(&self) -> Option<ConsensusState> {
+        match self.proposal {
+            Some(ref p) => Some(ConsensusState {
+                height: p.height,
+                round: p.round,
+                step: 3,
```

Comment:
Nit: I think it would be better to use constants / enums for this. See https://github.com/tendermint/tendermint/blob/411bc5e49fcc3dda866ae799ba2ceda0084f80d4/privval/file.go#L17-L35